purchases/src/main/kotlin/com/revenuecat/purchases/common/diagnostics/DiagnosticsTracker.kt

```diff
@@ -199,6 +202,36 @@ internal class DiagnosticsTracker(
         trackEvent(event)
     }
 
+    // region Offline Entitlements
+
+    fun trackEnteredOfflineEntitlementsMode() {
+        val event = DiagnosticsEntry(
+            name = DiagnosticsEntryName.ENTERED_OFFLINE_ENTITLEMENTS_MODE,
+            properties = mapOf(),
+        )
+        trackEvent(event)
+    }
+
+    fun trackErrorEnteringOfflineEntitlementsMode(error: PurchasesError) {
+        val isOneTimePurchaseFoundError = error.code == PurchasesErrorCode.UnsupportedError &&
+            error.underlyingErrorMessage == OfflineEntitlementsStrings.OFFLINE_ENTITLEMENTS_UNSUPPORTED_INAPP_PURCHASES
```

Comment:
I thought about making this a parameter instead of relying on processing the error... But this cleans it up a bit IMO.